tpu/src/test/java/tpu/QueuedResourceIT.java

```diff
@@ -16,57 +16,89 @@
 
-  public static void requireEnvVar(String envVarName) {
-    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
-        .that(System.getenv(envVarName)).isNotEmpty();
-  }
+  @Test
+  public void testCreateQueuedResourceWithSpecifiedNetwork() throws Exception {
+    TpuClient mockTpuClient = mock(TpuClient.class);
+    try (MockedStatic<TpuClient> mockedTpuClient = mockStatic(TpuClient.class)) {
+      mockedTpuClient.when(() -> TpuClient.create(any(TpuSettings.class)))
+          .thenReturn(mockTpuClient);
+
+      OperationFuture mockFuture = mock(OperationFuture.class);
+      when(mockTpuClient.createQueuedResourceAsync(any(CreateQueuedResourceRequest.class)))
+          .thenReturn(mockFuture);
+      CreateQueuedResourceWithNetwork.createQueuedResourceWithNetwork(
+          PROJECT_ID, ZONE, QUEUED_RESOURCE_NAME, NODE_NAME,
+          TPU_TYPE, TPU_SOFTWARE_VERSION, NETWORK_NAME);
 
-  @BeforeAll
-  public static void setUp() {
-    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
-    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+      verify(mockTpuClient, times(1))
+          .createQueuedResourceAsync(any(CreateQueuedResourceRequest.class));
+    }
   }
 
-  @AfterAll
-  public static void cleanup() {
-    DeleteForceQueuedResource.deleteForceQueuedResource(PROJECT_ID, ZONE, QUEUED_RESOURCE_NAME);
+  @Test
+  public void testGetQueuedResource() throws IOException {
+    GetQueuedResource mockGetQueuedResource = mock(GetQueuedResource.class);
+    QueuedResource mockQueuedResource = mock(QueuedResource.class);
+    try (MockedStatic<TpuClient> mockedTpuClient = mockStatic(TpuClient.class)) {
+      mockedTpuClient.when(TpuClient::create).thenReturn(mock(TpuClient.class));
+      when(mock(TpuClient.class)
+          .getQueuedResource(any(QueuedResourceName.class))).thenReturn(mockQueuedResource);
+
+      GetQueuedResource.getQueuedResource(PROJECT_ID, ZONE, NODE_NAME);
+
+      verify(mockGetQueuedResource, times(1))
+          .getQueuedResource(PROJECT_ID, ZONE, NODE_NAME);
+    }
   }
 
   @Test
-  public void testCreateQueuedResourceWithSpecifiedNetwork() throws Exception {
+  public void testDeleteTpuVm() {
+    TpuClient mockTpuClient = mock(TpuClient.class);
+    try (MockedStatic<TpuClient> mockedTpuClient = mockStatic(TpuClient.class)) {
+      mockedTpuClient.when(() -> TpuClient.create(any(TpuSettings.class)))
+          .thenReturn(mockTpuClient);
 
-    QueuedResource queuedResource = CreateQueuedResourceWithNetwork.createQueuedResourceWithNetwork(
-        PROJECT_ID, ZONE, QUEUED_RESOURCE_NAME, NODE_NAME,
-        TPU_TYPE, TPU_SOFTWARE_VERSION, NETWORK_NAME);
+      OperationFuture mockFuture = mock(OperationFuture.class);
+      when(mockTpuClient.deleteQueuedResourceAsync(any(DeleteQueuedResourceRequest.class)))
+          .thenReturn(mockFuture);
+      DeleteForceQueuedResource.deleteForceQueuedResource(PROJECT_ID, ZONE, QUEUED_RESOURCE_NAME);
```

Comment:
please add assert to verify that the return value is one that is expected to be returned by `deleteQueuedResourceAsync()` call or whatever the code sample method returns.

Reply:
Since we are mocking TpuClient, we can expect that the method has been called but no resources have been created before, so the return value of GetTpuVm.getTpuVm() will be null. Based on this example https://github.com/GoogleCloudPlatform/java-docs-samples/blob/f89f6faf150a64aacc49f9e4683c68a3c5000d7b/managedkafka/examples/src/test/java/examples/TopicsTest.java#L64 we can check that a String is printed when the method is running. Should I add a print like this to verify this? If we want to do E-2-E testing, we can move on to creating a real TpuClient and real resources.